### PR TITLE
Pgbouncer mode for PostgreSQL

### PR DESCRIPTION
**Pull request by pimeys**
Introducing a new flag for PostgreSQL: `pgbouncer`.

When `true`, after `BEGIN` of each transaction a query of `DEALLOCATE ALL` is executed to clean existing prepared statements from the session to prevent collisions.

To be able to use Quaint with pgbouncer's transaction mode, every query needs to be inside a transaction and the connection string should have `pgbouncer=true` set.

### CODE REVIEW

File: src/connector/postgres.rs [hunk 1]
```diff
@@ -223,9 +225,15 @@ impl PostgresUrl {
...
         let mut connect_timeout = None;
+        let mut pg_bouncer = false;
 
         for (k, v) in url.query_pairs() {
             match k.as_ref() {
+                "pgbouncer" => {
+                    pg_bouncer = v
+                        .parse()
+                        .map_err(|_| Error::builder(ErrorKind::InvalidConnectionArguments).build())?;
+                }
```
Comment (janpio) [inline src/connector/postgres.rs:236]: Does this check for `true`/`false` or just presence of `pgbouncer´? (Me no Rust...)

Comment (pimeys) [reply src/connector/postgres.rs:236]: true/false

Comment (pimeys) [reply src/connector/postgres.rs:236]: so `pg_bouncer` variable is a boolean, `parse` will parse whatever type is on the left side from the string.

File: src/pooled.rs [hunk 2]
```diff
@@ -51,6 +51,12 @@
 //! - `connect_timeout` defined in seconds (default: 5). Connecting to a
 //!   database will return a `ConnectTimeout` error if taking more than the
 //!   defined value.
+//! - `pgbouncer` either `true` or `false`. If set, allows usage with the
+//!   pgBouncer connection pool in transaction mode. Additionally a transaction
+//!   is required for every query for the mode to work. When starting a new
+//!   transaction, a deallocation query `DEALLOCATE ALL` is executed right after
+//!   `BEGIN` to avoid possible collisions with statements created in other
+//!   sessions.
```
Comment (janpio) [inline src/pooled.rs:59]: +1

**Pull request merged**